src/mito2/src/sst/parquet/writer.rs

```diff
@@ -125,4 +127,31 @@ impl ParquetWriter {
     }
 }
 
+#[derive(Default)]
+struct SourceStats {
+    /// Number of rows fetched.
+    num_rows: usize,
+    /// Min timestamp from fetched batches.
+    min_timestamp: Option<Timestamp>,
+    /// Max timestamp from fetched batches.
+    max_timestamp: Option<Timestamp>,
```

Comment:
Maybe merge `min_timestamp` and `max_timestamp` into `time_range: Option<(Timestamp, Timestamp)>` as these two field can only be either all present or not.